consensus/safety-rules/src/tests/suite.rs

```diff
@@ -565,47 +560,14 @@ fn test_sign_proposal_with_invalid_qc(safety_rules: &Callback) {
         key.as_ref(),
     );
     let err = safety_rules
-        .sign_proposal(a3.block().block_data().clone())
+        .sign_proposal(a3.block().block_data())
         .unwrap_err();
     assert_eq!(
         err,
         Error::InvalidQuorumCertificate("Fail to verify QuorumCert".into())
     );
 }
 
-fn test_sign_proposal_with_early_preferred_round(safety_rules: &Callback) {
-    let (mut safety_rules, signer, key) = safety_rules();
-
-    let (proof, genesis_qc) = test_utils::make_genesis(&signer);
-    let round = genesis_qc.certified_block().round();
-    safety_rules.initialize(&proof).unwrap();
-
-    let a1 = test_utils::make_proposal_with_qc(round + 1, genesis_qc, &signer, key.as_ref());
-    safety_rules
-        .sign_proposal(a1.block().block_data().clone())
-        .unwrap();
-
-    // Update preferred round with a few legal proposals
-    let a2 = make_proposal_with_parent(round + 2, &a1, None, &signer, key.as_ref());
-    let a3 = make_proposal_with_parent(round + 3, &a2, None, &signer, key.as_ref());
-    let a4 = make_proposal_with_parent(round + 4, &a3, Some(&a2), &signer, key.as_ref());
-    safety_rules.construct_and_sign_vote(&a2).unwrap();
-    safety_rules.construct_and_sign_vote(&a3).unwrap();
-    safety_rules.construct_and_sign_vote(&a4).unwrap();
-
-    let a5 = make_proposal_with_qc_and_proof(
-        round + 5,
-        test_utils::empty_proof(),
-        a1.block().quorum_cert().clone(),
-        &signer,
-        key.as_ref(),
-    );
-    let err = safety_rules
-        .sign_proposal(a5.block().block_data().clone())
-        .unwrap_err();
-    assert_eq!(err, Error::IncorrectPreferredRound(0, 2));
```

Comment:
seems like this should still fail, no? since the parent block is old, I wish we had better documented this test case...

Reply:
sign_proposal doesn't matter, the proposer is free to sign any garbage, the safety critical piece is upon voting.